contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -977,19 +975,26 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
                 );
             }
 
+            // How many times has this token id been used to commit to offers in the group?
+            uint256 commitCount = lookups.conditionalCommitsByTokenId[_tokenId][_groupId];
+
+            require(commitCount < _condition.maxCommits, MAX_COMMITS_TOKEN_REACHED);
+
             allow = holdsSpecificToken(_buyer, _condition, _tokenId);
 
             if (allow) {
                 // Increment number of commits to the group for this token id if they are allowed to commit
                 lookups.conditionalCommitsByTokenId[_tokenId][_groupId] = ++commitCount;
             }
         } else if (_condition.method == EvaluationMethod.Threshold) {
+            require(_condition.tokenType == TokenType.MultiToken ? _tokenId > 0 : _tokenId == 0, INVALID_TOKEN_ID);
```

Comment:
This is not backwards compatible - there might already exist conditions with `condition.tokenType == TokenType.MultiToken` and `_tokenId == 0`.